client/state/sites/selectors/is-site-on-woo-express-trial-period.ts

```diff
@@ -0,0 +1,15 @@
+import { PLAN_ECOMMERCE_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { getPurchases } from 'calypso/state/purchases/selectors';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Checks whether the current site is on the Woo Trial period.
+ *
+ * @param {AppState} state Global state tree
+ * @returns {?boolean} Returns true if the site is on the trial period
+ */
+export default function isSiteOnWooExpressTrialPeriod( state: AppState ) {
```

Comment:
Some general thoughts here:
1. It may be simpler to call this `isSiteOnWooExpressTrial()` - the use of `Period` feels like it implies that we should be checking the expiration of the plan as well
2. It may be worth borrowing logic similar to the code in [the isSiteOnPaidPlan() selector](https://github.com/Automattic/wp-calypso/blob/c8fb9b13f62074a61430f6fc84076e688ef5c0a4/client/state/selectors/is-site-on-paid-plan.js#L13), which relies on the `getCurrentPlan()` selector and the `isFreePlan()` function from `@automattic/calypso-products` -- we'd need an `isEcommerceTrialPlan()` helper function for that approach.
3. Might this selector make more sense in `client/state/sites/plans/selectors/`?